sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/ingestion/models/one/CommonSchemaLogSerializerTest.java

```diff
@@ -114,10 +114,19 @@ public void serializeAndDeserialize() throws JSONException {
         /* Check baseData and baseDataType from Part B not read into Part C. */
         log.getData().getProperties().put("baseDataType", "custom");
         log.getData().getProperties().put("baseData", new JSONObject());
-        Log copy = serializer.deserializeLog(serializer.serializeLog(log));
+        Log copy = serializer.deserializeLog(serializer.serializeLog(log), MockCommonSchemaLog.TYPE);
         log.getData().getProperties().remove("baseData");
         log.getData().getProperties().remove("baseDataType");
         assertEquals(log, copy);
+
+        /* Check deserialize fails without data type */
```

Comment:
`@Test(expected = JSONException .class)` doesn't work?

Reply:
We want to verify line 124 throws the exception, if it fails earlier because bug in the test due to future changes, it would appear green for an invalid reason.

Reply:
Can you comment on test method?